components/widgets/tag/tag.tsx

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {memo, MouseEventHandler, useMemo} from 'react';
+import styled, {css} from 'styled-components';
+import classNames from 'classnames';
+import {IconGlyphTypes} from '@mattermost/compass-icons/IconGlyphs';
+import glyphMap from '@mattermost/compass-icons/components';
+
+export type TagVariant = 'info' | 'success' | 'warning' | 'danger';
+
+export type TagSize = 'xs' | 'sm' | 'md' | 'lg'
+
+type Props = {
+    text: React.ReactNode;
+    uppercase?: boolean;
+    icon?: IconGlyphTypes;
+    variant?: TagVariant;
+    size?: TagSize;
+    onClick?: MouseEventHandler;
+    className?: string;
+};
+
+type TagWrapperProps = Required<Pick<Props, 'uppercase'>>;
+
+const TagWrapper = styled.div<TagWrapperProps>`
+    --tag-bg: var(--semantic-color-general);
+    --tag-bg-opacity: 0.08;
+    --tag-color: var(--semantic-color-general);
+
+    appearance: none;
+
+    display: inline-flex;
+    align-items: center;
+    align-content: center;
+    align-self: center;
+    gap: 4px;
+    max-width: 100%;
+    margin: 0;
+    overflow: hidden;
+
+    border: none;
+    border-radius: 4px;
+
+    font-family: 'Open Sans', sans-serif;
+    font-weight: 600;
+    line-height: 16px;
+    ${({uppercase}) => (
+        uppercase ? css`
+            letter-spacing: 0.02em;
+            text-transform: uppercase;
+        ` : css`
+            text-transform: none;
+        `
+    )}
+
+    &.Tag--xs {
+        height: 16px;
+        font-size: 10px;
+        line-height: 12px;
+        padding: 1px 4px;
+    }
+
+    &.Tag--sm {
+        height: 20px;
+        font-size: 12px;
+        line-height: 16px;
+        padding: 2px 5px;
+    }
+
+    &.Tag--md {
+        height: 24px;
+        font-size: 14px;
+        line-height: 20px;
+        padding: 2px 5px;
+    }
+
+    &.Tag--lg {
+        height: 28px;
+        font-size: 16px;
+        line-height: 22px;
+        padding: 2px 5px;
+    }
+
+    &.Tag--info,
+    &.Tag--success,
+    &.Tag--warning,
+    &.Tag--danger {
+        --tag-bg-opacity: 1;
+        --tag-color: 255, 255, 255;
+    }
+
+    &.Tag--info {
+        --tag-bg: var(--semantic-color-info);
+    }
+
+    &.Tag--success {
+        --tag-bg: var(--semantic-color-success);
+    }
+
+    &.Tag--warning {
+        --tag-bg: var(--semantic-color-warning);
+    }
+
+    &.Tag--danger {
+        --tag-bg: var(--semantic-color-danger);
+    }
+
+    background: rgba(var(--tag-bg), var(--tag-bg-opacity));
+    color: rgb(var(--tag-color));
+
+    ${({onClick}) => typeof onClick === 'function' && (
```

Comment:
its just a safety measure, since short circuit will return the value if it is falsy, so for instance when `onClick` is `0` it will just return `0` and might potentially break. This is not the case for boolean values